src/Demos/HSP.lagda

```diff
@@ -1,104 +1,81 @@
-version of this paper,~\cite{DeMeo:2021}, presents the formal proof in its entirety.}
-%and, apart from a few dozen imports from the \agdastdlib, is self-contained.}
+version~\cite{DeMeo:2021} presents the full proof.}
 \else
-We include here every line of code of our formal proof of Birkhoff's theorem
+We include here every line of code of our proof of Birkhoff's theorem
 in a single \agda module.  Apart from a few dozen imports from the \agdastdlib, the module
-is self-contains.
+is self-contained.
 \fi
 
-In the course of this presentation we highlight some of the challenging aspects of
-formalizing the basic definitions and theorems of universal algebra in type theory.
-%In particular, we touch on some technical issues that arise when attempting
-%to constructively prove basic results of our field.  Nonetheless,
-One positive contribution of this project is that it lends support to the claim that
-dependent type theory and the \agda language, despite the technical demands they place on
-the user, are accessible to working mathematicians (such as ourselves) who possess
-sufficient patience and resolve to codify their work in type theory in order to formally verify
-their results with a proof assistant.
+We highlight some of the challenging aspects of formalizing universal algebra in type theory.
+On the positive side, the patient mathematician with enough resolve and the will to learn
+dependent type theory can reap the rewards of the further insights that mechanization provide.
+% One positive contribution of this project is that it lends support to the claim that
+% dependent type theory and the \agda language, despite the technical demands they place on
+% the user, are accessible to working mathematicians (such as ourselves) who possess
+% sufficient patience and resolve to codify their work in type theory in order to formally verify
+% their results with a proof assistant.
 
 \ifshort\else
-Our presentation gives a sobering glimpse of the technical hurdles that must be overcome
-to conduct research in mathematics using dependent type theory and the \agda language.
-Nonetheless we hope our work does not discourage anyone from investing in these
-technologies and we remain committed to the use and promotion of type theory and proof
-assistants in general and in our own research. Indeed, we are excited to share the
-gratifying outcomes and achievements that resulted from attaining some degree of mastery
-of type theory, interactive theorem proving, and the \agda language.
+We give a sobering glimpse of the technical hurdles that must be overcome
+to conduct research in mathematics using dependent type theory in \agda.
+The results are rather gratifying and enlightening, and we feel are worth
+the investment. Users of Coq and Lean have recently reported similar feelings
+from the outcome of their formalization efforts.
 \fi
 
 \section{Preliminaries}
 
+We assume that the reader is familiar with \mltt, and can decipher its encoding in \agda.
```

Comment:
I would like to add "For the most part..." at the start of this first sentence of the Preliminaries section, to give us a little bit of leeway in case we want to explain something that expert Agda users would likely already know.  (I'm thinking mostly about the unabridged version here.)